presto-iceberg/src/main/java/com/facebook/presto/iceberg/PrestoIcebergNestedField.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.common.type.Type;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrestoIcebergNestedField
+{
+    private final String isOptional;
+    private final int id;
+    private final String name;
+    private final Type prestoType;
+    private final String doc;
+
+    @JsonCreator
+    public PrestoIcebergNestedField(
+            @JsonProperty("isOptional") String isOptional,
+            @JsonProperty("id") int id,
+            @JsonProperty("name") String name,
+            @JsonProperty("prestoType") Type prestoType,
+            @JsonProperty("doc") String doc)
+    {
+        this.isOptional = requireNonNull(isOptional, "isOptional is null");
+        this.id = id;
+        this.name = requireNonNull(name, "name is null");
+        this.prestoType = requireNonNull(prestoType, "prestoType is null");
+        this.doc = doc;
```

Comment:
requireNonNull? If this can be null, I would make the field an `Optional`

Reply:
I will make it Optional, it can be null based on the iceberg library.